src/components/CurrencyInputPanel/SwapCurrencyInputPanel.tsx

```diff
@@ -241,8 +278,21 @@ export default function SwapCurrencyInputPanel({
     setModalOpen(false)
   }, [setModalOpen])
 
+  const handleDisabledNumericalInputClick = useCallback(() => {
+    if (numericalInputSettings?.disabled) {
+      setAnimateShake(true)
+      setTimeout(() => setAnimateShake(false), ms('4s')) // reset shake animation state after 4s
+      numericalInputSettings.onDisabledClick()
+    }
+  }, [numericalInputSettings])
+
   const chainAllowed = isSupportedChain(chainId)
 
+  const [animateShake, setAnimateShake] = useState(false)
```

Comment:
can you rename this to something like `tooltipVisible` ? naming it after the animation made it harder to understand why we need this in state, as a reader